src/disco/bank/fd_bank_abi.c

```diff
@@ -266,10 +265,82 @@ is_upgradeable_loader_present( fd_txn_t * txn, uchar * payload, sanitized_txn_ab
   return 0;
 }
 
+extern int
+fd_ext_bank_load_account( void const *  bank,
+                          int           fixed_root,
+                          uchar const * addr,
+                          uchar **      owner,
+                          uchar *       data,
+                          ulong *       data_sz );
+
+int
+fd_bank_abi_resolve_address_lookup_tables( void const *     bank,
+                                           int              fixed_root,
+                                           ulong            slot,
+                                           fd_txn_t const * txn,
+                                           uchar const *    payload,
+                                           fd_acct_addr_t * out_lut_accts ) {
+  ulong writable_idx = 0UL;
+  ulong readable_idx = 0UL;
+  for( ulong i=0UL; i<txn->addr_table_adtl_cnt; i++ ) {
+    fd_txn_acct_addr_lut_t const * lut = &fd_txn_get_address_tables_const( txn )[ i ];
+    uchar const * addr = payload + lut->addr_off;
+
+    uchar * owner;
+    uchar   data[ 1UL+56UL+256UL*32UL ];
+    ulong   data_sz = sizeof(data);
+    int result = fd_ext_bank_load_account( bank, fixed_root, addr, &owner, data, &data_sz );
+    if( FD_UNLIKELY( result ) ) return FD_BANK_ABI_TXN_INIT_ERR_ACCOUNT_NOT_FOUND;
+
+    result = memcmp( owner, fd_solana_address_lookup_table_program_id.key, 32UL );
+    if( FD_UNLIKELY( result ) ) return FD_BANK_ABI_TXN_INIT_ERR_INVALID_ACCOUNT_OWNER;
+
+    if( FD_UNLIKELY( (data_sz<56UL) | (data_sz>(56UL+256UL*32UL)) ) ) return FD_BANK_ABI_TXN_INIT_ERR_INVALID_ACCOUNT_DATA;
+
+    fd_address_lookup_table_state_t table[1];
+    fd_bincode_decode_ctx_t bincode = {
+      .data    = data,
+      .dataend = data+data_sz,
+      .valloc  = {0},
+    };
+
+    result = fd_address_lookup_table_state_decode( table, &bincode );
+    if( FD_UNLIKELY( result!=FD_BINCODE_SUCCESS ) ) return FD_BANK_ABI_TXN_INIT_ERR_INVALID_ACCOUNT_DATA;
+
+    result = fd_address_lookup_table_state_is_lookup_table( table );
+    if( FD_UNLIKELY( !result ) ) return FD_BANK_ABI_TXN_INIT_ERR_ACCOUNT_UNINITIALIZED;
+
+    if( FD_UNLIKELY( (data_sz-56UL)%32UL ) ) return FD_BANK_ABI_TXN_INIT_ERR_INVALID_ACCOUNT_DATA;
+
+    ulong addresses_len = (data_sz-56UL)/32UL;
+    fd_acct_addr_t const * addresses = fd_type_pun_const( data+56UL );
+
+    ulong deactivation_slot = table->inner.lookup_table.meta.deactivation_slot;
+    if( FD_UNLIKELY( deactivation_slot!=ULONG_MAX && (deactivation_slot+512UL)<slot ) ) return FD_BANK_ABI_TXN_INIT_ERR_ACCOUNT_NOT_FOUND;
```

Comment:
I'm not an expert on this, but I don't think this handles skipped slots properly. Since this is only the leader pipeline, it's not the end of the world if it rejects a tiny fraction of transactions that reference an address lookup table deactivated between 512 and 512*(1+skip rate) slots ago, but it at least needs a comment to that effect